Graphviz fix

GNroy opened the pull request:
# What does this PR do ?

- Fixes graphviz cropped plots for notebooks
- Fixes duplicated Megatron-LM installation
- Fixes k2 library for compatibility with latest PyTorch

**Collection**: [ASR]

# Changelog 
- Now graphviz is built and installed from sources.

# Usage
* You can potentially add a usage example below

```python
# Add a code snippet demonstrating how to use this 
```

# Before your PR is "Ready for review"
**Pre checks**:
- [ ] Make sure you read and followed [Contributor guidelines](https://github.com/NVIDIA/NeMo/blob/main/CONTRIBUTING.md)
- [ ] Did you write any new necessary tests?
- [ ] Did you add or update any necessary documentation?
- [ ] Does the PR affect components that are optional to install? (Ex: Numba, Pynini, Apex etc)
  - [ ] Reviewer: Does the PR have correct import guards for all optional libraries?
  
**PR Type**:
- [ ] New Feature
- [x] Bugfix
- [ ] Documentation

If you haven't finished some of the above items you can still open "Draft" PR.


## Who can review?

Anyone in the community is free to review the PR once the checks have passed. 
[Contributor guidelines](https://github.com/NVIDIA/NeMo/blob/main/CONTRIBUTING.md) contains specific people who can review PRs to various areas.

# Additional Information
* Related to # (issue)

artbataev reviewed (dismissed):
Thanks for these fixes! I tested Docker build - everything is fine.

titu1994 commented on Dockerfile:
```diff
@@ -41,14 +41,28 @@ RUN apt-get update && \
   libavdevice-dev && \
   rm -rf /var/lib/apt/lists/*
+# install required libs for building graphviz
+RUN apt-get update && \
```
So much stuff for one library? What is graphviz used for? Plus, Why not just merge this above ?

titu1994 replied:
Plus don't install libpython or it's dev or any such thing. It interferes with the installed python already on the container

titu1994 commented on Dockerfile:
```diff
@@ -41,14 +41,28 @@ RUN apt-get update && \
-  git checkout ab0336a5c8eab77aa74ae604ba1e73decbf6d560 && \
-  pip install -e .
-
 # Install megatron core, this can be removed once 0.3 pip package is released
```
Commit id doesn't match above

artbataev replied:
There are 2 installation of Megatron-LM in main: with `ab0336a5c8eab77aa74ae604ba1e73decbf6d560` and `375395c187ff64b8d56a1cd40572bc779864b1bd`.
Which one is correct?
https://github.com/NVIDIA/NeMo/blob/main/Dockerfile#L47
https://github.com/NVIDIA/NeMo/blob/main/Dockerfile#L54

titu1994 commented on Dockerfile:
```diff
@@ -98,8 +112,18 @@ RUN pip install triton==2.0.0.dev20221202
-# install k2, skip if installation fails
 COPY scripts /tmp/nemo/scripts/
+# install graphviz, skip if installation fails
+RUN INSTALL_MSG=$(/bin/bash /tmp/nemo/scripts/installers/install_graphviz.sh --docker); INSTALL_CODE=$?; \
```
What is graphviz for ?

GNroy commented:
Hi @titu1994, I've addressed your comments.
`graphviz` is for WFST visualization (k2, pynini, ...)
Its binary does not display plots correctly in notebooks.
I've added an ok-ish `graphviz` installation in Dockerfile and requirements, and the correct one (implying build from source) for `REQUIRE_K2=true`.

artbataev reviewed (approved):
LGTM!

titu1994 reviewed (changes requested):
Needs discussing about license (non standard), necessity (why is a graph visualization library a required dependency for K2 ??) And whether it's realistic to bloat the apt get section wth so many libraries.

GNroy commented:
The licence is `Eclipse Public License - v 1.0`: https://gitlab.com/graphviz/graphviz/-/blob/main/epl-v10.txt (weak copyleft).

GNroy commented:
Hi all!
I just re-opened this PR and addressed comments of @titu1994.
The legal has approved general use of graphviz (the container use included).

titu1994 commented:
It's not been discussed yet - why is a graph visualization library being installed in the container ?

GNroy commented:
> why is a graph visualization library being installed in the container?

Simply for convenient prototyping. I always work inside NeMo containers (maybe I'm not the only one in that) and I'd like to have graphviz already installed rather than having to manually install it every time I run a new image.

titu1994 reviewed (approved):
Thank you for all the clarification and legal review. Looks good to merge
